suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/VisitorStateModifications.java

```diff
@@ -21,29 +21,39 @@
 import com.google.errorprone.BugPattern.SeverityLevel;
 import com.google.errorprone.VisitorState;
 import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.CompilationUnitTree;
 import com.sun.source.util.TreePath;
+import com.sun.source.util.TreePathScanner;
+import java.util.HashSet;
 import java.util.Optional;
 import java.util.Set;
 import java.util.logging.Logger;
 import java.util.stream.Stream;
 
 // CHECKSTYLE:ON
 
+@SuppressWarnings("RestrictedApi")
 public final class VisitorStateModifications {
     private static final Logger log = Logger.getLogger(VisitorStateModifications.class.getName());
-
     private static final ReportedFixCache FIXES = new ReportedFixCache();
+    private static final Set<CompilationUnitTree> SEEN = new HashSet<>();
```

Comment:
This `HashSet` will make strong reference on each `CompilationUnitTree` which will cause a memory leak. We should make this a weak set instead. There is no `WeakSet` type but you can use `Collections.newSetFromMap(new WeakHashMap<>())` to make one.